lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -157,15 +158,24 @@ def queue_job(self, job_wrapper):
         job = Job(self._pykube_api, k8s_job_obj)
         try:
             job.create()
-            job_id = job.metadata.get('name')
-            # define job attributes in the AsyncronousJobState for follow-up
-            ajs.job_id = job_id
-            # store runner information for tracking if Galaxy restarts
-            job_wrapper.set_external_id(job_id)
-        except Exception as e:
-            log.exception("Error occurred while creating kubernetes job object")
-            raise e
+        except HTTPError:
+            job_wrapper.fail("Kubernetes failed to create job.", exception=True)
+            log.exception("Kubernetes failed to create job, HTTP exception encountered")
+            ajs.runner_state = JobState.runner_states.UNKNOWN_ERROR
+            self.mark_as_failed(ajs)
+            return
+        if not job.name:
+            job_wrapper.fail("Kubernetes failed to create job.")
+            log.exception(f"Kubernetes failed to create job, empty name encountered: [{job.obj}]")
+            ajs.runner_state = JobState.runner_states.UNKNOWN_ERROR
+            self.mark_as_failed(ajs)
```

Comment:
Those direct actions should probably be handled by the work_queue
```suggestion
            self.work_queue.put((self.mark_as_failed, ajs))
```

Reply:
```python
    def mark_as_failed(self, job_state):
        self.work_queue.put((self.fail_job, job_state))
```
Given this function definition I thought it was appropriate to call directly. Do you disagree?